pkg/specgen/generate/namespaces.go

```diff
@@ -450,8 +454,13 @@ func specConfigureNamespaces(s *specgen.SpecGenerator, g *generate.Generator, rt
 		// the user or if we are creating a new UTS namespace.
 		// TODO: Should we be doing this for pod or container shared
 		// namespaces?
+		// we do not want to add domainname to the hostname, we need to keep track of it for later
 		g.SetHostname(hostname)
 	}
+
+	if s.Domainname != "" {
+		g.AddLinuxSysctl("kernel.domainname", domainname)
+	}
```

Comment:
your change for domainname got into the runtime-specs.  Why do not we use that (and also add support for it to crun)?

Reply:
ok, sounds good. There was initial discussion of getting this version in, but now that the spec PR has merged, I think we can wait on this.